system-helper/flatpak-system-helper.c

```diff
@@ -217,6 +284,24 @@ flatpak_invocation_return_error (GDBusMethodInvocation *invocation,
     }
 }
 
+static OngoingPull *
+take_ongoing_pull_by_dir (const gchar *src_dir)
+{
+  OngoingPull *pull = NULL;
+
+  G_LOCK (ongoing_pulls);
+  for (guint i = 0; i < ongoing_pulls->len; i++)
+    {
+      pull = g_ptr_array_index (ongoing_pulls, i);
+      if (g_strcmp0 (pull->src_dir, src_dir) == 0)
+        break;
+    }
+  g_ptr_array_remove (ongoing_pulls, pull);
```

Comment:
This is wrong in the case where `src_dir` is not found in the array: the last element in the array will be taken instead.